src/core/include/openvino/core/any.hpp

```diff
@@ -210,14 +218,36 @@ struct Read<
     std::map<K, T, C, A>,
     typename std::enable_if<std::is_default_constructible<K>::value && std::is_default_constructible<T>::value>::type> {
     void operator()(std::istream& is, std::map<K, T, C, A>& map) const {
-        while (is.good()) {
-            std::string str;
-            is >> str;
-            auto k = from_string<K>(str);
-            is >> str;
-            auto v = from_string<T>(str);
-            map.emplace(std::move(k), std::move(v));
+        char c;
+
+        is >> c;
+        OPENVINO_ASSERT(c == '{', "Failed to parse std::map<K, T>. Starting symbols is not '{', it's ", c);
+
+        while (c != '}') {
```

Comment:
What if we check the incorrect map: `{1, 2, 3, }`?
Should we check for `\0` also?

Reply:
We don't plan to implement generic parser which handles all the errors.
Parsing is required only because of mix new / old APIs on plugin API level.
E.g. user specified new properties, which are converted internally to strings to be passed to old LoadNetwork call, later it's parsed back to ov::Any.

So, once plugins migrate to new API, we can remove parsing capabilities.